compiler/src/iree/compiler/Codegen/LLVMCPU/test/set_transform_strategy.mlir

```diff
@@ -0,0 +1,64 @@
+        %1 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%c0) : !flow.dispatch.tensor<writeonly:tensor<24x512x16x1xf32>>
+        %2 = flow.dispatch.tensor.load %0, offsets = [0, 0], sizes = [384, 512], strides = [1, 1] : !flow.dispatch.tensor<readonly:tensor<384x512xf32>> -> tensor<384x512xf32>
+        %3 = tensor.empty() : tensor<24x512x16x1xf32>
+        %pack = tensor.pack %2 inner_dims_pos = [0, 1] inner_tiles = [16, 1] into %3: tensor<384x512xf32> -> tensor<24x512x16x1xf32>
+        flow.dispatch.tensor.store %pack, %1, offsets = [0, 0, 0, 0], sizes = [24, 512, 16, 1], strides = [1, 1, 1, 1] : tensor<24x512x16x1xf32> -> !flow.dispatch.tensor<writeonly:tensor<24x512x16x1xf32>>
+        return
+      }
+    }
+  }
+}
+// CHECK-DAG:   #[[TRANSLATION:.+]] = #iree_codegen.translation_info<TransformDialectCodegen>
+// CHECK:         hal.executable.export public @pack
+// CHECK-SAME:       translation_info = #[[TRANSLATION]]
+// CHECK-LABEL: func.func @pack
+// CHECK:         transform.sequence
+// CHECK:         %[[PACK:.+]] = transform.structured.match ops{["tensor.pack"]} in %arg0
+// CHECK:         %[[FORALL:.+]], %[[TILED_PACK:.+]] = transform.iree.tile_to_forall_and_workgroup_count_region
+// CHECK-SAME:      %[[PACK:.+]] num_threads [1]
+// CHECK:         %[[CAST:.+]] = cast %[[TILED_PACK]] : !pdl.operation to !transform.op<"tensor.pack">
+// CHECK:         %[[PAD:.+]], %[[RESHAPE:.+]], %[[TRANS:.+]] =
+// CHECK-SAME:      transform.structured.lower_pack %[[CAST]]
+// CHECK:         %[[TILED_TRANS:.+]], %{{.+}} = transform.structured.tile %[[TRANS]][1, 8, 8]
+// CHECK:         %[[CAST:.+]] = cast %[[TILED_TRANS]] : !transform.op<"linalg.transpose"> to !pdl.operation
+// CHECK:         %[[GEN:.+]] = transform.structured.generalize %[[CAST]]
+// CHECK:         %[[FUNC:.+]] = transform.structured.match ops{["func.func"]} in %arg0
+// CHECK:         transform.iree.apply_patterns %[[FUNC]]
+// CHECK-SAME:      rank_reducing_linalg_via_reshapes
+// CHECK:         %{{.*}} = transform.structured.vectorize
+// CHECK:         %{{.*}} = transform.structured.hoist_redundant_tensor_subsets
+// CHECK:         %{{.*}} = transform.iree.bufferize %arg0
+// CHECK:         transform.iree.erase_hal_descriptor_type_from_memref
+// CHECK:         transform.iree.forall_to_workgroup
+// CHECK:         %{{.*}} = transform.vector.apply_transfer_permutation_patterns
+// CHECK:         %{{.*}} = transform.vector.split_transfer_full_partial
+// CHECK:         %{{.*}} = transform.vector.transfer_to_scf
+// CHECK:         %{{.*}} = transform.vector.lower_transfer
+// CHECK:         %{{.*}} = transform.vector.lower_shape_cast
+// CHECK:         %{{.*}} = transform.vector.lower_transpose
+// CHECK-SAME:      lowering_strategy = shuffle
+// CHECK-SAME:      avx2_lowering_strategy = true
```

Comment:
ah nice you are using the special microkernel asm I added a while back, great to see this finally used :)